specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+        });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    }
+
+    return 0;
+}
+```
+
+## Unpackaged app Registering and handling a notification
+
+Similar to packaged apps, unpackaged apps can also Register to send app notifications and act as
+targets for activation events. There is one basic difference in the Registration for
+Unpackaged apps versus their packaged counterparts: they will need to specify their own friendly App
+DisplayName and IconUri since they don't have an app manifest file that contains the same.
+
+```cpp
+int main()
+{
+    PCWSTR const c_iconName = L"icon.png";
+    std::filesystem::path binaryPath = wil::GetModuleFileNameW<std::wstring>(nullptr);
+    std::filesystem::path directoryPath = binaryPath.parent_path();
+    std::filesystem::path iconPath = directoryPath / std::filesystem::path(c_iconName);
+
+    Uri iconUri(iconPath.wstring().c_str());
+    AppNotificationActivationInfo activationInfo(L"SampleApp", iconUri);
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in FG. So intercept toast activators via FG event
+        const auto token = AppNotificationManager::Default().AppNotificationActivated([](const auto&, const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+            {
+                ProcessNotificationArgs(notificationActivatedEventArgs);
+            });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    }
+
+    return 0;
+}
+```
+
+## Displaying an App Notification
+
+To display a notification, an app needs to define a payload using an xml schema. In addition, the developer can set a tag and group property which allows the app to replace a prior notification with the same tag and group with an updated version of the payload.
+Finally, the ExpiresOnReboot property can also be set to ensure that the notification is automatically deleted from
+the Action Centre after device reboots.
+
+```cpp
+// Display a notification in action centre using a tag and group
+void DisplayNotification(const winrt::Windows::Data::Xml::Dom::XmlDocument& doc)
```

Comment:
Does the API rely on `XmlDocument? Even if it's only sample code, we should avoid this type since most developers will be using a different XML parser (or just strings) this is a problem. Moving data between different parsers is inefficient.

Reply:
This is a short term problem we have that we are planning to rectify with a "toast builder" API in the future.

Reply:
Tying this API to the WinRT XML DOM is a big mistake. AFAICT the implementation in Windows just serializes the DOM and then uses XmlLite to parse it -- meaning we could have the WinAppSDK deal in Strings only. Down-level it turns into a DOM to pass to the OS (which then turns it back into a String) but up-level we can update the OS to rely on Strings and save everyone effort.

Reply:
RECOMMEND: Just take a string.

/* static AppNotification.CreateFromXml(string xml); */

AppNotification(String contentXml);